### PR TITLE
Enhanced the functionality of getSSMLResponse

### PR DESCRIPTION
**Pull request by mandnyc**
I noticed there was no .gitignore file for this project so I created one that excludes a few common ignorable files. E.g  WebStorm's .idea and Mac's .DS_STORE file. Keep up the great work! :)

### CODE REVIEW

**Comment by mandnyc** [comment]
## Enhanced the functionality of getSSMLResponse

This function can now accept different types of values. Prior, it only accepted strings, numbers and booleans.

In addition to accepting strings, numbers, and booleans, the developer can now pass in an object that provides the type (SSML or PlainText) and speech properties. In the case of bad input, I throw some exceptions that detail why the values were invalid. e.g., functions, symbols (ES6), or objects without a type and speech property.

When an object is passed that has a type and speech property, this method handles case sensitivity. e.g., "ssml", "Ssml", etc. will be converted to "SSML". Same holds true to "PlainText".

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.idea/
```
Comment (deegles) [inline .gitignore]: We prefer to use global .gitignore files

**Comment by deegles** [comment]
Could you please remove the .gitignore file?

**Comment by defel** [comment]
Hi, 

first great to see this PR! But it is really sad that it is open since more than one month, without any review or merge. 

So, I will try to give you some feedback about this PR.

File: lib/response.js [hunk 2]
```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
...
+    if (messageType === "object") {
+        if (message.type && message.speech) {
+            if (isValidSpeechType(message.type)) {
+                return {
+                    type: normalizeSpeechType(message.type),
+                    speech: message.speech
+                };
+            }
+        } else {
+            throw new Error('The message must be an object with properties "type" and "speech".');
```
Comment (defel) [inline lib/response.js]: Throwing Errors mean they have to been catched later? This should be done in App- or Library-Code?

File: lib/response.js [hunk 3]
```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
...
+ *
+ * @param message The message to convert into a object that has type + speech properties.
+ * @returns {{type: string, speech: string}}
+ */
 function getSSMLResponse(message) {
-    return {
-        type: 'SSML',
-        speech: `<speak> ${message} </speak>`
-    };
+    if (message === null || message === undefined) {
```
Comment (defel) [inline lib/response.js]: Could be simplefied with: `if (!message) {`

File: lib/response.js [hunk 4]
```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
...
+        throw new Error('The message must not be null or undefined');
+    }
+
+    var messageType = typeof(message);
+    if (messageType === "function") {
```
Comment (defel) [inline lib/response.js]: Is it really necessary to check for every possible type here? As this type of errors are compile-time checks, the default errors should be ok.

File: lib/response.js [hunk 5]
```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
...
+        throw new Error('The message must not be a function');
+    }
+
+    if (messageType === "object") {
+        if (message.type && message.speech) {
+            if (isValidSpeechType(message.type)) {
```
Comment (defel) [inline lib/response.js]: no else condition? (What happens if messageType is an object with an invalid `speechType`?

(Currently the error `The message must not be of type: object` is shown, which is not correct.

File: lib/response.js [hunk 6]
```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
...
+        }
+    }
+    
+    if (isAcceptableTypes(messageType)) {
+        return {
+            type: 'SSML',
+            speech: `<speak>${message}</speak>`
+        };
+    }
+    throw new Error("The message must not be of type: " + messageType + ". Valid types are string, number, boolean and object (with type/speech properties)."); // This addresses Symbols in ES6.
```
Comment (defel) [inline lib/response.js]: Again, throwing errors for compile-time error / type checks is IMHO not a good way to handle this type of errors in javascript. 

Maybe something like https://flowtype.org/ can handle this job better?

File: lib/response.js [hunk 7]
```diff
@@ -196,10 +196,86 @@ function buildSpeechletResponse(options) {
...
+ *
+ * The message passed in can be one of the followings:
+ *
+ * - string, number or boolean which will be converted to the following response:
+ *  { type: 'SSML', speech: '<speak>{message value}</speak>' }
+ *
+ * - when an object is passed in that has type and speech properties, it'll just be returned.
+ *
+ * @param message The message to convert into a object that has type + speech properties.
+ * @returns {{type: string, speech: string}}
```
Comment (defel) [inline lib/response.js]: Good comment 👍

**Comment by tianrenz** [comment]
Hi @mandnyc , 

Just checking if this PR is still active. It looks like there are some merge conflicts with this PR. If you could resolve these conflicts, we would be happy to review it now. Apologies for not getting back to you in time before. 

At the meantime, ASK SDK v2 for Node.js launched last month with many design changes aiming to resolve many issues reported on v1 SDK through the years. We would encourage you to check it out [here](https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs). You can find the documentation in the wiki section. 

Regards

**Pull request closed without merging**